src/main/java/com/avispl/symphony/dal/communicator/aver/ptz/AverPTZCommunicator.java

```diff
@@ -0,0 +1,361 @@
+	 *
+	 * @param cameraID the {@code java.lang.String} field
+	 */
+	public void setCameraID(String cameraID) {
+		this.cameraID = cameraID;
+	}
+
+	/**
+	 * Retrieves {@code {@link #panSpeed}}
+	 *
+	 * @return value of {@link #panSpeed}
+	 */
+	public String getPanSpeed() {
+		return panSpeed;
+	}
+
+	/**
+	 * Sets {@code panSpeed}
+	 *
+	 * @param panSpeed the {@code java.lang.String} field
+	 */
+	public void setPanSpeed(String panSpeed) {
+		this.panSpeed = panSpeed;
+	}
+
+	/**
+	 * Retrieves {@code {@link #tiltSpeed}}
+	 *
+	 * @return value of {@link #tiltSpeed}
+	 */
+	public String getTiltSpeed() {
+		return tiltSpeed;
+	}
+
+	/**
+	 * Sets {@code tiltSpeed}
+	 *
+	 * @param tiltSpeed the {@code java.lang.String} field
+	 */
+	public void setTiltSpeed(String tiltSpeed) {
+		this.tiltSpeed = tiltSpeed;
+	}
+
+	/**
+	 * This method is recalled by Symphony to control specific property
+	 *
+	 * @param controllableProperty This is the property to be controlled
+	 */
+	@Override
+	public void controlProperty(ControllableProperty controllableProperty) {
+
+	}
+
+	/**
+	 * This method is recalled by Symphony to control a list of properties
+	 *
+	 * @param controllableProperties This is the list of properties to be controlled
+	 */
+	@Override
+	public void controlProperties(List<ControllableProperty> controllableProperties) {
+
+	}
+
+	/**
+	 * This method is recalled by Symphony to get the list of statistics to be displayed
+	 *
+	 * @return List<Statistics> This return the list of statistics.
+	 */
+	@Override
+	public List<Statistics> getMultipleStatistics() {
+		return Collections.emptyList();
+	}
+
+	/**
+	 * This method is used to read data from device
+	 *
+	 * @param command This is a byte array of command to check done reading or not
+	 * @return byte[] This returns the response receive from device
+	 */
+	@Override
+	protected byte[] read(byte[] command) throws IOException {
+		String responseString;
+		boolean hasACK = false;
+		byte[] response;
+
+		do {
+			response = super.read(command);
+			responseString = getHexByteString(response);
+
+			// If this is ACK packet, need to save to the flag to check done reading
+			if (responseString.endsWith(getHexByteString(ReplyStatus.ACK.getCode()))) {
+				hasACK = true;
+			}
+
+		} while (!doneReading(command, responseString, hasACK));
+
+		return response;
+	}
+
+	/**
+	 * This method is used to check when done reading
+	 *
+	 * @param command This is a command parameter for method read executed before throw error (if it has ACK)
+	 * @param responseString This is a string represented for response to be checked
+	 * @param hasACK This is a boolean value to check if it has ACK packet or not (ACK -> ERROR -> COMPLETION, if it has error, need to read COMPLETION packet left)
+	 * @return boolean This is a boolean value return if done reading or not
+	 */
+	public boolean doneReading(byte[] command, String responseString, boolean hasACK) throws CommandFailureException, IOException {
```

Comment:
This is a kind of complex logic. I think we should have the UT for it

Reply:
Ok I will add it